### PR TITLE
Bundle images of twemoji (resolve Issues#61)

### LINKED ISSUE #61: Emoji isn't displayed on enviroment can't connect internet.
Emoji isn't displayed on enviroment can't connect internet.

### PR DESCRIPTION
**Pull request by yasumichi**
To display emoji on enviroment can't connect internet, bundle 72x72 images of twemoji. 

More Detail in Japanese, See  http://qiita.com/yasumichi/items/73370ba740ef4fa6e2ac .

### New files

- images/twemoji/LICENSE-GRAPHICS
- images/twemoji/72x72/*.png (872 files)

### Updated scripts

To change twemoji.base from CDN to local path, updated bellow scripts.

- coffee/classes/mds_markdown.coffee
```diff
diff --git a/coffee/classes/mds_markdown.coffee b/coffee/classes/mds_markdown.coffee
index 5288f52..f09fa2b 100644
--- a/coffee/classes/mds_markdown.coffee
+++ b/coffee/classes/mds_markdown.coffee
@@ -93,6 +93,7 @@ module.exports = class MdsMarkdown
   constructor: (settings) ->
     opts         = extend({}, MdsMarkdown.default.options, settings?.options || {})
     plugins      = extend({}, MdsMarkdown.default.plugins, settings?.plugins || {})
+    twemoji.base = settings?.twemojiBase || twemoji.base
     @afterRender = settings?.afterRender || null
     @markdown    = MdsMarkdown.createMarkdownIt.call(@, opts, plugins)
     @afterCreate()
```
- coffee/slide.coffee
```diff
diff --git a/coffee/slide.coffee b/coffee/slide.coffee
index 7d49aaa..33bbcfe 100644
--- a/coffee/slide.coffee
+++ b/coffee/slide.coffee
@@ -13,7 +13,8 @@ document.addEventListener 'DOMContentLoaded', ->
       for target in $(@).attr('data-marp-path-resolver').split(/\s+/)
         $(@).attr(target, resolvePathFromMarp($(@).attr(target)))
 
-    Markdown = new clsMarkdown({ afterRender: clsMarkdown.generateAfterRender($) })
+    twemojiBasePath = Path.resolve(__dirname, '../images/twemoji/') + '/'
+    Markdown = new clsMarkdown({ afterRender: clsMarkdown.generateAfterRender($), twemojiBase: twemojiBasePath })
 
     themes = {}
     themes.current = -> $('#theme-css').attr('href')
```

### CODE REVIEW

File: coffee/classes/mds_markdown.coffee [hunk 1]
```diff
@@ -93,6 +93,7 @@ module.exports = class MdsMarkdown
   constructor: (settings) ->
     opts         = extend({}, MdsMarkdown.default.options, settings?.options || {})
     plugins      = extend({}, MdsMarkdown.default.plugins, settings?.plugins || {})
+    twemoji.base = settings?.twemojiBase || twemoji.base
```
Comment (yhatt) [inline coffee/classes/mds_markdown.coffee]: Well probably the base-path won't change, but I think it should not set as twemoji's common setting when it has per `MdsMarkdown` instances.

And `MdsMarkdown` class has its `@default` variable for putting default options. I wish to set local path as a `@default.twemoji.base`.

File: coffee/slide.coffee [hunk 2]
```diff
@@ -13,7 +13,8 @@ document.addEventListener 'DOMContentLoaded', ->
       for target in $(@).attr('data-marp-path-resolver').split(/\s+/)
         $(@).attr(target, resolvePathFromMarp($(@).attr(target)))
 
-    Markdown = new clsMarkdown({ afterRender: clsMarkdown.generateAfterRender($) })
+    twemojiBasePath = Path.resolve(__dirname, '../images/twemoji/') + '/'
```
Comment (yhatt) [inline coffee/slide.coffee]: ...and the separator should use `Path.sep` instead of `/` to output correct path in Windows.

Comment (yhatt) [inline coffee/slide.coffee]: Oh, it requires overriding [twemoji's callback](https://github.com/twitter/twemoji/blob/gh-pages/README.md#callback) to return correctly path in Windows.

**Comment by yhatt** [comment]
Based on above comments, How about this?

- coffee/classes/mds_markdown.coffee

```diff
diff --git a/coffee/classes/mds_markdown.coffee b/coffee/classes/mds_markdown.coffee
index 5288f52..5bbc1e5 100644
--- a/coffee/classes/mds_markdown.coffee
+++ b/coffee/classes/mds_markdown.coffee
@@ -34,6 +34,10 @@ module.exports = class MdsMarkdown
         shortcuts: {}
       'markdown-it-katex': {}

+    twemoji:
+      callback: (icon, opts) -> "#{opts.base}#{opts.size}#{Path.sep}#{icon}#{opts.ext}"
+      base: Path.resolve(__dirname, '../../images/twemoji/') + Path.sep
+
   @createMarkdownIt: (opts, plugins) ->
     md = markdownIt(opts)
     md.use(require(plugName), plugOpts ? {}) for plugName, plugOpts of plugins
@@ -89,10 +93,13 @@ module.exports = class MdsMarkdown
   rulers: []
   settings: new MdsMdSetting
   afterRender: null
+  twemojiOpts: {}

   constructor: (settings) ->
     opts         = extend({}, MdsMarkdown.default.options, settings?.options || {})
     plugins      = extend({}, MdsMarkdown.default.plugins, settings?.plugins || {})
+    @twemojiOpts = extend({}, MdsMarkdown.default.twemoji, settings?.twemoji || {})
+
     @afterRender = settings?.afterRender || null
     @markdown    = MdsMarkdown.createMarkdownIt.call(@, opts, plugins)
     @afterCreate()
@@ -106,8 +113,8 @@ module.exports = class MdsMarkdown
       html_block: rules.html_block

     extend rules,
-      emoji: (token, idx) ->
-        twemoji.parse(token[idx].content)
+      emoji: (token, idx) =>
+        twemoji.parse(token[idx].content, @twemojiOpts)

       hr: (token, idx) =>
         ruler.push token[idx].map[0] if ruler = @_rulers
```

`coffee/slide.coffee` has not changed.

I thank for your contribution! :heart_eyes:

**Comment by yasumichi** [comment]
I thanks for advice looks better.

I updated that's files on your advice, and pushed to my repository.

**Pull request merged**